src/pip/_internal/cli/main_parser.py

```diff
@@ -57,6 +99,27 @@ def parse_command(args: List[str]) -> Tuple[str, List[str]]:
     #  args_else: ['install', '--user', 'INITools']
     general_options, args_else = parser.parse_args(args)
 
+    # --python
+    if general_options.python and "_PIP_RUNNING_IN_SUBPROCESS" not in os.environ:
```

Comment:
Can we instead pop out the `--python` argument from the sys.argv?

There's only two variants -- it's either `["--python=value"]` or `["--python", "value"]` and both of those can be trimmed instead of setting an environment variable.

Reply:
Hmm, I did think about that[^1]. There are three main problems I see:

1. It's all too possible that someone later adds a `-P` alias and forgets to add it to the argv stripping code. Then boom.
2. People could do all sorts of weird things - what about setting the `PIP_PYTHON` environment variable? I think global options have env variable / config file equivalents, don't they? That would also cause things to go boom.
3. The worst point for me is that the failure mode is so bad. We worked on forkbombs when we implemented build isolation, and it's really bad - basically it locked up my PC so badly that a power off was the only fix. I really don't want a problem with the `--python` option to risk triggering that...

I'm willing to consider it, but I'd want someone else to do the testing of what could go wrong on their machine 😉

[^1]: Technically I thought about `--python value`, I forgot `--python=value`, there's your first mistake already 🙂

Reply:
OK, you're right and this does seem like a good way to do it. The only other option would be to check if we're in the pip runner, but that feels... highly coupled.